src/cardano/CardanoEngine.ts

```diff
@@ -566,6 +567,77 @@ export class CardanoEngine extends CurrencyEngine<
       publicAddress: bech32Address
     }
   }
+
+  getStakeAddress = async (): Promise<string> => {
+    const { bech32Address } = asSafeCardanoWalletInfo(this.walletInfo).keys
+
+    const paymentAddress = Cardano.Address.from_bech32(bech32Address)
+
+    // Get the stake credential
+    const baseAddr = Cardano.BaseAddress.from_address(paymentAddress)
+
+    if (baseAddr == null) {
+      throw new Error("This address doesn't have staking rights")
+    }
+
+    const stakeCredential = baseAddr.stake_cred()
+
+    // Create a stake address from the credential
+    const stakeAddress = Cardano.RewardAddress.new(
+      paymentAddress.network_id(),
+      stakeCredential
+    )
+
+    // Convert to bech32
+    const bech32StakeAddress = stakeAddress.to_address().to_bech32()
+
+    return bech32StakeAddress
+  }
+
+  decodeStakingTx = async (encodedTx: string): Promise<EdgeTransaction> => {
+    const { bech32Address } = asSafeCardanoWalletInfo(this.walletInfo).keys
+
+    const tx = Cardano.Transaction.from_hex(encodedTx)
+    const txHash = Cardano.hash_transaction(tx.body())
+
+    const txJson = tx.to_js_value()
+
+    // Validate the transaction is a staking transaction:
```

Comment:
Personal preference: I would put the actual comparison of the addresses in outputs and our bech32address in this function instead of buried in a cleaner

Reply:
Interesting preference. My thought is that asStakingTx.ts is concerned with validation, so why move that concern up to the caller?